src/model.hpp

```diff
@@ -208,7 +208,8 @@ class model {
         current_spikes().get().push_back({source, tspike});
     }
 
-    void attach_sampler(cell_member_type probe_id, sampler_function f, time_type tfrom = 0) {
+    template <typename F>
+    void attach_sampler(cell_member_type probe_id, F&& f, time_type tfrom = 0) {
```

Comment:
I am using `std::forward` because I just can't be bothered reverse engineering how samplers work another time. When these are lowered into the back end, they need to be documented.